backend/src/email/middlewares/email-transactional.middleware.ts

```diff
@@ -23,53 +26,120 @@ export const InitEmailTransactionalMiddleware = (
   authService: AuthService
 ): EmailTransactionalMiddleware => {
   const logger = loggerWithLabel(module)
+  interface ReqBody {
+    subject: string
+    body: string
+    from: string
+    recipient: string
+    reply_to: string
+    attachments?: { data: Buffer; name: string }[]
+  }
 
-  async function sendMessage(
+  async function saveMessage(
     req: Request,
     res: Response,
     next: NextFunction
   ): Promise<void> {
-    const ACTION = 'sendMessage'
+    const action = 'saveMessage'
+    logger.info({ message: 'Saving email', action })
+    const {
+      subject,
+      body,
+      from,
+      recipient,
+      reply_to: replyTo,
+      attachments,
+    }: ReqBody = req.body
 
-    try {
-      const {
+    const { fromName, fromAddress } = parseFromAddress(from)
+    const userEmail = (await authService.findUser(req.session?.user?.id))?.email
+    const emailMessageTx = await EmailMessageTx.create({
+      userEmail,
+      fromName,
+      fromAddress,
+      recipient,
+      params: {
         subject,
         body,
         from,
-        recipient,
-        reply_to: replyTo,
-        attachments,
-      } = req.body
+        replyTo,
+      },
+      hasAttachment: !!attachments,
```

Comment:
why don't we save the list of file names instead of a flag?

Reply:
Just saw this! Following today's discussion, thinking we should store this as a [JSON array element](https://www.postgresql.org/docs/11/functions-json.html#:~:text=Get-,JSON%20array%20element,-(indexed%20from%20zero) with the following shape:

```TS
type Attachments = AttachmentMetadata[]

interface AttachmentMetadata {
  filename: string
  author: string
  hash: string
  // not sure what other stuff to include; will play by ear
}
```

It's my first time using JSON array element, but I think should be fine?

Reply:
Yup should be fine, they will be saved as a jsonb string. Checkout `params` from `email-message` model for similar implementation

Reply:
hmm I think it's a bit different, since it's a JSON array rather than a JSON? because there could be multiple attachments? 🤔

Reply:
I'm thinking of sth like `{ type: Data.Type.Array(DataType.JSON) }` — I'll experiment and make sure it works before requesting for review :)